server/apiv2/handlers/members.go

```diff
@@ -0,0 +1,215 @@
+
+		// Get etcd ID by name.
+		id, err := getMemberIDByName(svr, c.Param("name"))
+		if err != nil {
+			c.AbortWithStatusJSON(http.StatusInternalServerError, err.Error())
+			return
+		}
+
+		if id == 0 {
+			c.AbortWithStatus(http.StatusNotFound)
+			return
+		}
+
+		if err := deleteMemberByID(svr, id); err != nil {
+			c.AbortWithStatusJSON(http.StatusInternalServerError, err.Error())
+			return
+		}
+		c.JSON(
+			http.StatusOK,
+			nil,
+		)
+	}
+}
+
+func getMemberIDByName(svr *server.Server, name string) (id uint64, err error) {
+	members, err := svr.GetMembers()
+	if err != nil {
+		return
+	}
+	for _, m := range members {
+		if name == m.Name {
+			id = m.GetMemberId()
+			break
+		}
+	}
+	return
+}
+
+func deleteMemberByID(svr *server.Server, id uint64) error {
+	// Delete config.
+	err := svr.GetMember().DeleteMemberLeaderPriority(id)
+	if err != nil {
+		return err
+	}
+
+	// Delete dc-location info.
+	err = svr.GetMember().DeleteMemberDCLocationInfo(id)
+	if err != nil {
+		return err
+	}
+
+	client := svr.GetClient()
+	_, err = etcdutil.RemoveEtcdMember(client, id)
+	if err != nil {
+		return err
+	}
+	return nil
+}
+
+func getAllMembers(svr *server.Server) (*pdpb.GetMembersResponse, error) {
+	req := &pdpb.GetMembersRequest{Header: &pdpb.RequestHeader{ClusterId: svr.ClusterID()}}
+	grpcServer := &server.GrpcServer{Server: svr}
+	members, err := grpcServer.GetMembers(context.Background(), req)
+	if err != nil {
+		return nil, err
+	}
+	dclocationDistribution, err := svr.GetTSOAllocatorManager().GetClusterDCLocationsFromEtcd()
+	if err != nil {
+		return nil, err
+	}
+	for _, m := range members.GetMembers() {
+		m.DcLocation = ""
+		binaryVersion, e := svr.GetMember().GetMemberBinaryVersion(m.GetMemberId())
+		if e != nil {
+			log.Error("failed to load binary version", zap.Uint64("member", m.GetMemberId()), errs.ZapError(e))
+		}
+		m.BinaryVersion = binaryVersion
+		deployPath, e := svr.GetMember().GetMemberDeployPath(m.GetMemberId())
+		if e != nil {
+			log.Error("failed to load deploy path", zap.Uint64("member", m.GetMemberId()), errs.ZapError(e))
+		}
+		m.DeployPath = deployPath
+		if svr.GetMember().GetEtcdLeader() == 0 {
+			log.Warn("no etcd leader, skip get leader priority", zap.Uint64("member", m.GetMemberId()))
+			continue
+		}
+		leaderPriority, e := svr.GetMember().GetMemberLeaderPriority(m.GetMemberId())
+		if e != nil {
+			log.Error("failed to load leader priority", zap.Uint64("member", m.GetMemberId()), errs.ZapError(e))
+			continue
+		}
+		m.LeaderPriority = int32(leaderPriority)
+		gitHash, e := svr.GetMember().GetMemberGitHash(m.GetMemberId())
+		if e != nil {
+			log.Error("failed to load git hash", zap.Uint64("member", m.GetMemberId()), errs.ZapError(e))
+			continue
+		}
+		m.GitHash = gitHash
+		found := false
+		for dcLocation, serverIDs := range dclocationDistribution {
+			for _, serverID := range serverIDs {
+				if serverID == m.MemberId {
+					m.DcLocation = dcLocation
+					found = true
```

Comment:
Too many `{}` in `{}`. how about define an independent function and make it friendly for our reviewer? For example replace L200-L213 as following:
```
          for dcLocation, serverIDs := range dclocationDistribution {
			found := contains(serverIDs, m.MemberId)
			if found {
				m.DcLocation = dcLocation
				break
			}
		}
```